lib/diagnostic.rb

```diff
@@ -91,13 +91,13 @@
 # In a Ruby string, write how you would remove the last two elements from this
 # array
 
-Diagnostic.remove_from_array = 'your answer here'
+Diagnostic.remove_from_array = 'arr.pop(2)'
 
 # Taking the result from the question above, what would `arr` now equal:
 arr[arr.length + 2] = 99
 # Write your response as a ruby string
 
-Diagnostic.array_add_to = 'your answer here'
+Diagnostic.array_add_to = 'I keep staring at this and not understanding what it is asking'
```

Comment:
It's asking what `arr` looks like after the statement `arr[arr.length + 2] = 99` is executed.  `arr[arr.length + 2] = 99` sets the element at the length of the array plus two to `99`.  In this case, `arr` was originally `[12, 34, 56, 67]`.  `arr.pop(2)` changes it to `[12, 34]`.  `arr[arr.length + 2] = 99` is equivalent to `arr[4] = 99`, which sets the fifth element to `99`.  After all of this is done, `arr` is `[12, 34, nil, nil, 99]`.